### PR TITLE
feat: cloud dialog component

### PR DESCRIPTION
**Pull request by luanvdw**
**_Work in progress_**

Closes [cloud#894](https://github.com/prisma/cloud/issues/894)

Cloud Dialog Component.

### CODE REVIEW

File: src/cloud/dialog/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  return (
+    <DialogContext.Provider value={value}>
+      <OverlayProvider>
+        <Button onPress={() => state.open()}>{title}</Button>
+        {state.isOpen && children}
+      </OverlayProvider>
+    </DialogContext.Provider>
+  )
+}
```
Comment (madebysid) [inline src/cloud/dialog/Dialog.tsx:146]: Can we move the DialogContainer definition above DialogBody so it is more readable?

File: src/cloud/dialog/Dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+  const { dialogProps, titleProps } = useDialog({ id, role: "dialog" }, ref)
+
+  return (
+    <OverlayContainer>
+      <FocusScope>
+        <div
+          className={cn("absolute", "px-6 py-4", "bg-white")}
+          style={{
+            left: dimmensions?.left ? dimmensions.left : 0,
```
Comment (madebysid) [inline src/cloud/dialog/Dialog.tsx:75]: I think you have this typo in this entire PR :D

dimmensions -> dimensions

File: src/cloud/dialog/Dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    title,
+    subtitle,
+    icon,
+    dimmensions: parentContainerRef?.current?.getBoundingClientRect(),
+    close: state.close,
+  }
+
+  return (
+    <DialogContext.Provider value={value}>
+      <OverlayProvider>
```
Comment (madebysid) [inline src/cloud/dialog/Dialog.tsx:140]: You don't need another `OverlayProvider`. All Lens components are wrapped in a `LensProvider`, which nests its children in an `OverlayProvider`

File: src/components/dialog/Dialog.tsx [hunk 4]
```diff
@@ -94,80 +118,108 @@ export type DialogBodyProps = {
...
       isDismissable: true,
       shouldCloseOnBlur: shouldCloseOnBlur,
       onClose: close,
     },
-    overlayRef
+    ref
   )
 
-  usePreventScroll()
+  !isEmbedded && usePreventScroll()
```
Comment (madebysid) [inline src/components/dialog/Dialog.tsx:141]: Doesn't this cause problems? Conditional hook and all?

Comment (madebysid) [reply src/components/dialog/Dialog.tsx:141]: Oh probably not, `isEmbedded` doesn't change between renders

File: src/cloud/dialog/Dialog.tsx (hunk 2 above)
Comment (luanvdw) [reply src/cloud/dialog/Dialog.tsx]: 🙈  lol thanks!

File: src/components/dialog/Dialog.tsx (hunk 4 above)
Comment (luanvdw) [reply src/components/dialog/Dialog.tsx:141]: Indeed, I did not have any problems locally. Will double check to make sure 👍

**Review by madebysid — APPROVED** [review]

**Pull request merged**